src/components/Popup/Popup.jsx

```diff
@@ -0,0 +1,30 @@
+import "./Popup.css";
+import React from 'react';
+import { FaCheckCircle } from "react-icons/fa";
+import { FaTimesCircle } from "react-icons/fa";
+import { AiFillExclamationCircle } from "react-icons/ai";
+
+export const Popup = React.forwardRef(({popupType, dialogueText, children}, ref) => {
+  const renderNotificationIcon = () => {
+    switch (popupType) {
+      case "success":
+        return <FaCheckCircle size="7rem" color="var(--clr-primary)"/>;
+      case "error":
+        return <FaTimesCircle size="7rem" color="var(--clr-error)"/>;
+      case "warning":
+        return <AiFillExclamationCircle size="7rem" color="var(--clr-warning)"/>;
+      default:
+        break;
+    }
+  };
+  
+  return (
+    <dialog ref={ref} className="popup">
```

Comment:
- HTML has made a new tag called `<dialog>`. The `<dialog>` has 2 innate functions, `showModal()` and `closeModal()`
- `ref` is React's way of allowing us to execute these innate functions
- When we wanna use these innate functions, we need to know the exact element to call them. In raw HTML, CSS, JS we would do
```
const dialogElement = getElementById("dialogBox");
dialogElement.showModal();
```
- In React, we just need to pass a `ref` (reference) to it and use it `ref.current.showModal()`. You can see this in action in `AddMember.jsx`